crates/sui-config/src/builder.rs

```diff
@@ -329,12 +329,20 @@ impl<R: rand::RngCore + rand::CryptoRng> ConfigBuilder<R> {
                     },
                 };
 
+                let anemo_config = {
+                    let mut config = anemo::Config::default();
+                    // Set a default size limit of 8 MiB for all RPCs
+                    config.max_frame_size = Some(8 << 20);
+                    config
+                };
+
```

Comment:
Instead of including this here, could we just override this value in sui-node/src/lib.rs?

Reply:
It didn't seem correct to me to be making changes at that point in the code to the `NodeConfig` it was given

Reply:
For the time being I think we should override this value there, even if it doesn't feel super great. The main reason for this is that all the nodes need to agree on this value for now in order for the system to work properly. Once we have a better idea of the limits we want to set at various places in the system then we can revisit this.

Reply:
There's a lot of stuff in `NodeConfig` that all nodes need to agree on for the system to work properly. Is there a reason this value is unique among those? It feels wrong to take a NodeConfig and then silently override only this one particular config value that's 3 layers deep, the kind of thing that would drive me absolutely crazy later on if I were trying to figure out why a config I'm setting isn't working like I expected.

Reply:
The main reason why this doesn't work today is that we have a number of different configs in different places which may not pick up this "default".

> It feels wrong to take a NodeConfig and then silently override only this one particular config value that's 3 layers deep, the kind of thing that would drive me absolutely crazy later on if I were trying to figure out why a value I'm setting isn't working like I expected.

How about a compromise then? Instead of overriding, i'm more concerned with this being a default in sui/narwhal. How about we set the value to 8MiB in sui-node iff the value wasn't previously set?

Reply:
Ok, modified as discussed